pkg/kf/commands/utils/prefix_filter.go

```diff
@@ -0,0 +1,34 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+)
+
+type prefixFilter struct {
+	nw newlineWriter
+}
+
+func NewPrefixFilter(prefixes map[string]io.Writer, defaultWriter io.Writer) io.Writer {
+	return &prefixFilter{
+		nw: newlineWriter{
+			dest: func(data []byte) (int, error) {
+				for prefix, dest := range prefixes {
+					idx := bytes.Index(data, []byte(prefix))
```

Comment:
```suggestion
					if bytes.HasPrefix(data, []byte(prefix)) {
						dest.Write(bytes.TrimPrefix(data, []byte(prefix)))
						dest.Write([]byte{'\n'})
					}
```

Alternatively, `bytes.SplitAfterN` could work if you can't guarantee the prefix exists at the start of the line.

Reply:
That looks more readable. Thanks!